### PR TITLE
view swapping, backable super frag

### LINKED ISSUE #95: View swapping / transitions infrastructure
-  [x] toolbar behavior (back on settings, hamburger on list, back on login)

-  [x] drawer

-  [x] settings modal

-  [x] lock screen

### LINKED ISSUE #96: Disable rotation

### PR DESCRIPTION
**Pull request by sashei**
resolves #95 
resolves #96

### CODE REVIEW

File: app/src/main/res/layout/include_backable.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+<android.support.v7.widget.Toolbar xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/toolbar"
+    android:layout_width="match_parent"
+    android:layout_height="?attr/actionBarSize"
+    android:background="@color/colorPrimary"
+    android:theme="@style/ThemeOverlay.AppCompat.ActionBar"
+    tools:showIn="@layout/fragment_setting" />
```
Comment (linuxwolf) [inline app/src/main/res/layout/include_backable.xml]: Is this reference to `@layout/fragment_setting` intentional?  Have any side-effects been observed?

Comment (linuxwolf) [reply app/src/main/res/layout/include_backable.xml]: Reading the [reference](https://developer.android.com/studio/write/tool-attributes#toolsshowin), it seems this only affects the Android Studio layout editor, to help with visually editing the toolbar. It's probably fine, but might be worth noting this.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -18,4 +18,6 @@
     <string name="hello_blank_fragment">This is a placeholder for an OAuth login to FxA</string>
     <string name="fxa_login_logmein_btn">Log Me In</string>
     <string name="settings_go_here">settings go here!</string>
+    <string name="placeholder_unlock_button">Unlock one\'s Firebox</string>
```
Comment (linuxwolf) [inline app/src/main/res/values/strings.xml]: ahem (-;

Comment (linuxwolf) [reply app/src/main/res/values/strings.xml]: Including the app_name is something we have to do by hand, in this `strings.xml`. )-:

File: app/src/main/res/layout/include_backable.xml (hunk 1 above)
Comment (jhugman) [reply app/src/main/res/layout/include_backable.xml]: 👍 

This is used in multiple places, so I've deleted the `showIn` attribute.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (jhugman) [reply app/src/main/res/values/strings.xml]: :+1:

Copied other resources.

**Pull request merged**